src/mesh/boundary_info.C

```diff
@@ -1504,6 +1533,25 @@ void BoundaryInfo::remove_side (const Elem * elem,
 {
   libmesh_assert(elem);
 
+#ifdef LIBMESH_ENABLE_AMR
+  // Here we have to stop and check if somebody tries to remove an ancestor's boundary ID
+  // through a child
+  if (elem->level() && _children_on_boundary)
```

Comment:
Let's not check _children_on_boundary here?  Trying to remove a child side even when no child sides have been added is still a case we want to scream about.

Reply:
Removed!